### ISSUE TITLE
Breadcrumb component does not provide the correct accessible name, role and values 

### ISSUE THREAD

**Issue by JoaoTMDias** [ant-design/ant-design#33821]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-breadcrumb-minimal-reproduction-tqjuk?file=/index.js)

### Steps to reproduce
* Open the link to minimal reproduction in a new window (https://tqjuk.csb.app/)
* Use a screen-reader (NVDA, Jaws or Voice Over) and try to encounter the breadcrumb landmark.
* Try to find how many breadcrumbs are there on the list of links. No screen reader will announce that.
* Try to also navigate through the list of breadcrumbs and find which one is the one that's active (or associated with the current page).

### What is expected?
According to the WAI-ARIA Authoring Practices 1.2, by using the correct name, role, and values for the breadcrumb design pattern, the component should be able to:
* Inform the accessibility tree of the existence of a navigation landmark pertaining to the breadcrumb.
* A screen-reader should then be able to read out loud the landmark label. (Ex:  "Breadcrumb. Navigation").
* The list of links (or other elements) that create the breadcrumb should belong to an ordered list
* That list should tell the screen-reader how many items (or levels, in this case) does the breadcrumb have. For example, if there are 4 links ( "Home > Products / Smeg / Kniffe Block"), the user should be able to infer that information since we know that the more links there are the deeper we are within a an architecture of pages.
* The last item of the list should indicate the element that represents the current item within a container or set of related elements. In this case, the page. That can be done using the "aria-current" attribute, with the value of "page".

### What is actually happening?
* Since the wrapping HTML element is a div - and thus conveys no semantic meaning - there is no default way of telling a screen-reader that, in fact, that wrapper element is part of a breadcrumb component. That could be done by using extra ARIA attributes to convey the semantic meaning that the nav element already does. For example, the role of "navigation" and an accessible label (either with "aria-label" or "aria-labelledby"). But that forces the user to do extra work to make something accessible that could be that way by default.
* If there is no HTML element telling the accessibility tree that there is a list of links, then any screen-reader won't be able to infer that those links are related to another and thus reads any of those as separate (e.g: "Kniffe Block. Link" instead of "Kniffe Block. Link. Item 4 of 4")
* The accessibility tree has no way of creating the reference that tells the screen-reader that the last link is, in fact, the one that is currently active and is thus related to the current page. 

| Environment | Info |
|---|---|
| antd | 4.18.4 |
| React | 17.0.1 |
| System | Windows 11 Pro Version	10.0.22000 Build 22000 |
| Browser | Microsoft Edge v97.0.1072.69 (Official build) (64-bit) |

---
I want to use the breadcrumb component for my company's design system, but I'm unable to do so unless there are some changes in the markup to fix these accessibility issues. (Will open a PR later with the fix)

**Comment by JoaoTMDias** [comment]
Hello! I can tackle this issue and will open a PR as soon as possible with the suggested fix for this, including updates to the documentation and related unit tests.

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/34082
